enterprise/kernel/src/test/java/org/neo4j/test/TestEnterpriseGraphDatabaseFactory.java

```diff
@@ -44,9 +44,37 @@
     protected GraphDatabaseBuilder.DatabaseCreator createDatabaseCreator( File storeDir,
                                                                           GraphDatabaseFactoryState state )
     {
-        return config -> {
-            config.put( "unsupported.dbms.ephemeral", "false" );
-            return new EnterpriseGraphDatabase( storeDir, config, state.databaseDependencies() );
+        return new GraphDatabaseBuilder.DatabaseCreator() {
+
+            @Override
+            public GraphDatabaseService newDatabase( Map<String,String> config )
+            {
+                config.put( "unsupported.dbms.ephemeral", "false" );
+                return new GraphDatabaseFacadeFactory( DatabaseInfo.ENTERPRISE, EnterpriseEditionModule::new )
```

Comment:
It's not easy to see what's being changed here. Also, `TestGraphDatabaseFactory` already has a `TestGraphDatabaseFacadeFactory` class that can maybe extended instead? It creates a `TestDatabasePlatformModule` that already does special stuff for logging.